### PR TITLE
Add a switch to allow turning on ValidationVisitor shortcircuiting

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs [hunk 1]
```diff
@@ -110,6 +110,12 @@ public class ValidationVisitor
         /// </summary>
         public bool ValidateComplexTypesIfChildValidationFails { get; set; }
 
+        /// <summary>
+        ///  Gets or sets a value that determines if <see cref="ValidationVisitor"/> can short-circuiting validation when a model
```
Comment (mkArtakMSFT) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs]: … can short-circuit validation...

File: src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs [hunk 2]
```diff
@@ -442,6 +442,19 @@ public int MaxModelValidationErrors
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value that determines if <see cref="ValidationVisitor"/>
+        /// can short-circuiting validation when a model does not have any associated validators.
```
Comment (mkArtakMSFT) [inline src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs]: same here ... short-circuit...

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs [hunk 3]
```diff
@@ -283,6 +289,23 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
...
+                // No validators will be created for this graph of objects. Mark it as valid if it wasn't previously validated.
+                var entries = ModelState.FindKeysWithPrefix(key);
+                foreach (var item in entries)
+                {
+                    if (item.Value.ValidationState == ModelValidationState.Unvalidated)
+                    {
+                        item.Value.ValidationState = ModelValidationState.Valid;
+                    }
+                }
+            }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs]: This code block duplicates line 276-271 - I'd expect to see a single line diff here (L295)

**Comment by pranavkm** [comment]
🆙 📅 - the flag is a compat switch that's on by default in 2.2

**Pull request merged**